main/src/ca/uwaterloo/flix/language/phase/LoadBytecode.scala

```diff
@@ -63,21 +63,43 @@ object LoadBytecode extends Phase[ExecutableAst.Root, ExecutableAst.Root] {
     * Finally, we generate bytecode for each name. We keep the types and names in an interfaces map, so that given
     * a closure's signature, we can lookup the functional interface it's called through.
     *
-    * 4. Generate and load bytecode.
+    * 4. Extract declared enums
+    * At this stage we traverse the definitions to extract all enum types. We can't use root.enums since the type
+    * of enum's field is not available there.
+    *
+    * 5. Create tag interface.
+    * At this step we generate an interface which declare a single method `getTag()` which returns an string
+    * containing the tag that each enum object represents. Every enum will implement this interface
+    *
+    * 6. Create an interface for each enum type.
+    * We generate an interface for each enum type and each enum case will implement this interface. This interface
+    * does not contain any method and it extends Tag interface
+    *
+    * 7. Create a class for each enum case.
+    * A class will be generated for each enum case.
+    * This class contains `tag` and `value` as its fields.
+    * It also includes implementations of methods such as `getTag()` which returns a String containing value of `tag`,
+    * `getValue()` which returns `value`, `hashCode()` which returns an Int containing hashCode of the case,
+    * `toString()` which returns a String of representation of the case and `equals(o: Object)` which returns `true`
+    * if the `o` is of type of enum case and the underlying value of both o and `this` is equal.
+    * This class implements the enum interface of its type
+    *
+    * 8. Generate and load bytecode.
     * As of this step, we have grouped the constants into separate classes, transformed non-functions into 0-arg
     * functions, collected all the declarations in a map, and created and loaded functional interfaces and collected
     * them in a map.
     * Now, for each class, we generate and load the bytecode. We also initialize the static Flix field to point to
     * `this`.
     *
-    * 5. Load the methods.
+    * 8. Load the methods.
     * For each constant, we use reflection to get the corresponding java.lang.reflect.Method object.
     * This is actually a bit tricky. We need the rewritten lambda types (non-functions -> 0-arg functions, free
     * variables eliminated) to perform the reflection lookup, so we iterate over constantsMap. However, we want to
     * mutate the original constant from root.constants, not the one in constantsMap (which will get GC'd).
     */
   def run(root: ExecutableAst.Root)(implicit flix: Flix): Validation[ExecutableAst.Root, CompilationError] = {
     implicit val _ = flix.genSym
+    val fixedPrefix = List("ca", "waterloo", "flix", "enums")
```

Comment:
I think we should put them in some other namespaces, but let us discuss tomorrow. It might even be possible to use the Flix namespace as the JVM namespace.